resolve-ref/action.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 Kinetic Commerce
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Resolve a ref-ish value
+description: |
+  Resolves a provided open pull request, branch, tag, or commit ref to
+  a valid commit reference.
+
+  Ref resolution is always in the order of `pr`, `branch`, `tag`, `commit`,
+  `default-branch`, `repo-default-branch`.
+author: 'KineticCafe'
+inputs:
+  token:
+    description: |
+      The GitHub token for this action.
+    required: true
+    default: ${{ github.token }}
+  ref:
+    description: |
+      The branch name, open pull request number, tag, or commit reference to
+      resolve.
+
+      Pull request numbers must always begin with `#` to distinguish them from
+      commits that are all decimal digits. Closed or merged pull requests will
+      be ignored.
+
+      If missing or empty, this action will fall back to the resolved value of
+      `default-branch`.
+    required: false
+  default-branch:
```

Comment:
(naming nit) The use of "default" here feels odd. As I don’t understand what resolve means as a term of software development “art” I can’t suggest a good replacement, maybe just `branch`?

This might just be me completely misunderstanding the whole context of the call to this action.

Reply:
The `inputs.ref` value is the PR, branch name, tag, or commit-ish git ref we want to resolve into a commit git ref. The `input.default-branch` is an optional fall-back branch name to resolve into a commit git ref. If neither is provided or both are blank (`''` and `null` are treated the same here) then we fall back to making an API call against the repo to get the "default" branch (e.g., `main` or `master` usually) and resolve *that* to a commit git ref.

So, `default-branch` is one of the better names for this. It could be `fallback-branch`, but it cannot just be `branch`.